### PR TITLE
Fixes to equality and hashing

### PR DESCRIPTION
**Pull request by eric-wieser**
Haven't implemented hashing on `pdop` etc, since that's more work.

I expect this to fix some LGTM warnings.

### CODE REVIEW

File: galgebra/mv.py [hunk 1]
```diff
@@ -436,18 +436,12 @@ def blade_rep(self):
...
-        if isinstance(A, Mv):
-            diff = (self - A).expand()
-            if diff.obj == S(0):
-                return False
-            else:
-                return True
+    def __hash__(self):
+        if self.is_scalar():
+            # ensure we match equality
+            return hash(self.obj)
```
Comment (utensil) [inline galgebra/mv.py]: :+1:

File: galgebra/ga.py [hunk 2]
```diff
@@ -370,10 +371,15 @@ def preset(setting, root='e', debug=False):
...
+
     def __eq__(self, ga):
-        if self.name == ga.name:
-            return True
-        return False
+        return self.name == ga.name
+
+    if sys.version_info.major < 3:
+        def __ne__(self, other):
+            return not (self == other)
```
Comment (eric-wieser) [inline galgebra/ga.py]: All this seems a bit dubious to me - my guess is that this is to support `copy.copy(ga)` and still have them be equal.

Comment (utensil) [reply galgebra/ga.py]: I can't follow. What do you mean?

Comment (eric-wieser) [reply galgebra/ga.py]: It's not clear to me why `__eq__` is overloaded in the first place (or why ga.name is needed).

For now, I will assume there is a good reason, and this patch just makes sure that everything is consistent.

I think we can consider the discussion about removing `Ga.name` (and by association these methods) as covered by #196.

**Comment by eric-wieser** [comment]
Now that gh-197 is merged, this change must go in, else the `ga != ga` lines in that PR will fail on python 2.7

File: galgebra/ga.py (hunk 2 above)
Comment (utensil) [reply galgebra/ga.py]: I think it's good to keep an bussiness/math equivalence beyond object sameness. Not just for technical reasons like copy.

Comment (eric-wieser) [reply galgebra/ga.py]: As an example of why the current approach is a bad idea, this does the wrong thing:
```python
import pickle
ga1 = Ga.build('e*1|2|3', g=[1,1,1])
ga2 = Ga.build('e*1|2', g=[1,1])
with open('bad.pkl', 'wb') as f:
    pickle.dump(f, ga2)
```
```python
import pickle
ga2_expected = Ga.build('e*1|2', g=[1,1])
ga2_bad = Ga.build('e*1|2|3', g=[1,1,1])
with open('bad.pkl', 'rb') as f:
    ga2 = pickle.load(f)
# both of these pass, but shouldn't
assert ga2_bad == ga2
assert ga2_expected != ga2
```
This is worse behavior than just using object sameness would be.

Comment (utensil) [reply galgebra/ga.py]: We could do the equivalence in some other way. I don't like the `.name` way either, but somehow feels there should be an bussiness/math equivalence.

**Pull request merged**